### PR TITLE
Adding ngrok Gateway API Implementation to Implementations page

### LINKED ISSUE #2842: Add ngrok to list of Gateway API implementations
**What would you like to be added**:
ngrok is working on an implementation for the Gateway API, we want to be added to the implementation list

**Why this is needed**:
ngrok is a popular service and we want to make sure users are able to find out about it and get the information they need to use it.

FYI: I am a PM at ngrok working on this initiative.

### PR DESCRIPTION
**Pull request by salilsub**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:
[ngrok](https://ngrok.com/) has created a Gateway API implementation. This is an entry for the implementations page listing it and providing additional information.
**Which issue(s) this PR fixes**:

Fixes #2842 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by youngnick** [comment]
Hi @salilsub, having ngrok grow Gateway API support sounds awesome, but we'll need you to do the CNCF CLA to accept the PR, sorry.

Also, in the links, the quickstart link for Gateway API currently points to an overview page for the Ngrok Ingress Controller, and I couldn't see any examples or instructions for Gateway API. Any chance those could be added?

I'd also encourage you to look into running the conformance test suite against your implementation and telling us if there's anything amiss.

**Comment by salilsub** [comment]
Hi @youngnick, I'm working through the CLA stuff right now (just filed with easyCLA)
With respect to the links, we're in the middle of re-working some of our docs and the link provided is instead going to be for a more general "Using ngrok and k8s" page. The repo will still be the same (our ingress controller and Gateway API implementation are in the same repo). I decided to open this PR since we are close to Kubecon Europe and I didn't know what kind of turnaround there would be on PR reviews
We are also working on the conformance testing/reports. I don't have the results on that right now but once I get them, I can add it in.

File: site-src/implementations.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ cover, and documentation to help users get started.
 - [Kong][10] (GA)
 - [Kuma][11] (beta)
 - [LiteSpeed Ingress Controller][19]
+- [ngrok][33] (preview)
 - [NGINX Gateway Fabric][12] (GA)
```
Comment (shaneutt) [inline site-src/implementations.md:34]: For neutrality we have historically done lexicographical ordering using the english alphabet, so this actually needs to go _after_ nginx:

```suggestion
- [NGINX Gateway Fabric][12] (GA)
- [ngrok][33] (preview)
```

File: site-src/implementations.md [hunk 2]
```diff
@@ -392,6 +394,21 @@ If you have any suggestions or experience issues with NGINX Gateway Fabric, plea
 [nginx-disc-new]:https://github.com/nginxinc/nginx-gateway-fabric/discussions/new
 [nginx-slack]:https://nginxcommunity.slack.com/channels/nginx-gateway-fabric
 
+
+### ngrok Kubernetes Operator
+
+[ngrok Kubernetes Operator][ngrok-k8s-operator] provides an implementation of the Gateway API that uses [ngrok's ingress-as-a-service][ngrok]. This project uses the Gateway API to support routing traffic from ngrok's global network to applications running on Kubernetes clusters. This easily adds the benefits of ngrok, like security, network policy, and a global presence with the simplicity of cloud service. The operator contains both a Gateway API implementation as well as a controller using Kubernetes Ingress. The Gateway API implementation is currently under development and supports only the Gateway, GatewayClass and HTTPRoute. As the TLSRoute and TCPRoute move from experimental to stable, they will also be implemented.
```
Comment (shaneutt) [inline site-src/implementations.md:400]: So it's possible I'm missing something, so please do correct me if I am: the title here is `ngrok Kubernetes Operator` however the links provided here are all to what appears to be called the `ngrok Kubernetes Ingress Controller`. I looked at the repository you provided and the documentation on your site and everything seems to be `Ingress Controller`. I couldn't find any relevant pages when searching "operator" in your docs site, and I also searched all your issues and PRs for the term "operator" and didn't find anything that stood out to me. The `README.md` for the repository also didn't have "operator" in it. Was this a typo? Was this meant to say `ngrok Kubernetes Ingress Controller`? OR is it the wrong link, are we missing a link to the right repository?

File: site-src/implementations.md [hunk 3]
```diff
@@ -392,6 +394,21 @@ If you have any suggestions or experience issues with NGINX Gateway Fabric, plea
...
+
+You can read our [docs][ngrok-k8s-gwapi-docs] or [quickstart guide][ngrok-k8s-gwapi-qs] for more information. If you have any feature requests or bug reports, please [create an issue][ngrok-issue-new]. You can also reach out for help on [Slack][ngrok-slack]
```
Comment (shaneutt) [inline site-src/implementations.md:402]: I followed these links and I could not find any information, or even reference to "Gateway API", am I missing something? Please let me know where the information relevant to your Gateway API implementation is.

File: site-src/implementations.md [hunk 4]
```diff
@@ -392,6 +394,21 @@ If you have any suggestions or experience issues with NGINX Gateway Fabric, plea
...
+
```
Comment (shaneutt) [inline site-src/implementations.md:403]:
```suggestion
```

File: site-src/implementations.md [hunk 5]
```diff
@@ -392,6 +394,21 @@ If you have any suggestions or experience issues with NGINX Gateway Fabric, plea
...
+
+[ngrok-k8s-operator]:https://github.com/ngrok/kubernetes-ingress-controller
```
Comment (shaneutt) [inline site-src/implementations.md:405]: Again, just wanting to check if there was a typo here? Wrong name, or wrong link?

File: site-src/implementations.md [hunk 6]
```diff
@@ -392,6 +394,21 @@ If you have any suggestions or experience issues with NGINX Gateway Fabric, plea
...
+[ngrok]:https://ngrok.com
+[ngrok-k8s-gwapi-docs]:https://ngrok.com/docs/k8s/
+[ngrok-k8s-gwapi-qs]:https://ngrok.com/docs/k8s/
```
Comment (shaneutt) [inline site-src/implementations.md:408]: These are both the same link, and neither seems to go to any Gateway API implementation information?

**Review by shaneutt — CHANGES_REQUESTED** [review]
This implementation is marked as `preview`, can you please help me to find the information about how users in the community can preview it? I didn't find the right docs, but perhaps I wasn't looking in the right places?

File: site-src/implementations.md (hunk 6 above)
Comment (salilsub) [reply site-src/implementations.md]: There was originally a plan to have separate links, decided to switch to just one so I removed this reference.

**Comment by salilsub** [comment]
@shaneutt based on your above comments, there are two open questions, naming and docs/links
### Naming of the repo
We originally supported only an ingress controller, but we have built our Gateway API implementation within the existing repo since there is a lot of overlap in the code. We're going to rename the repo, but we want to make sure we don't break anything anywhere before we do that rename. We know links from docs to the repo should redirect but we're not sure about some other things and want to verify before renaming. My plan was to submit a second PR to update the link to the repo once we did that.

### Docs/Links
We also have a docs PR (https://github.com/ngrok/ngrok-docs/pull/639) that I am currently updating. The plan is to merge tomorrow in advance of Kubecon. This has information about Gateway API and the page in the links for this PR should be the same because we are providing a single page describing both the existing ingress controller and Gateway API implementation with options to select which one they want to use.

**Comment by shaneutt** [comment]
Gotcha, so should we wait for some of this other work to complete? Or what are your hopes/expectations here?

**Comment by salilsub** [comment]
@shaneutt I would prefer if we merge the changes in this PR now since the only open item is the repo rename, the docs changes were merged yesterday. Once we rename the repo, all the links should redirect until I can submit another PR in the future.

**Review by shaneutt — APPROVED** [review]
Ok cool, thanks for the clarifications and the updates. I looked over it again and the new docs seem apropos.

/ok-to-test
/approve

and /cc @youngnick for an LGTM :+1:

**Comment by salilsub** [comment]
@youngnick wanted to follow up on this.

**Comment by youngnick** [comment]
Thanks for the ping, @salilsub, this LGTM now.

/lgtm

**Comment by Success411** [comment]
Round 3 of the day INSTANT 20pips✅

Let’s CLOSE our profit now and set breakeven if you wish to hold now‼️

We focus on scalping traders🔥🔥🔥

**Pull request merged**